explainers/navigation-to-unsigned-bundles.md

```diff
@@ -0,0 +1,363 @@
+  - [Service Workers](#service-workers)
+  - [URLs for bundle components](#urls-for-bundle-components)
+- [Open design questions](#open-design-questions)
+  - [Network access](#network-access)
+- [Security and privacy considerations](#security-and-privacy-considerations)
+- [Considered alternatives](#considered-alternatives)
+  - [Alternate formats considered](#alternate-formats-considered)
+    - [Save as directory tree](#save-as-directory-tree)
+    - [Save as MHTML](#save-as-mhtml)
+    - [Save as Web Archive](#save-as-web-archive)
+    - [WARC](#warc)
+    - [Mozilla Archive Format](#mozilla-archive-format)
+    - [A bespoke ZIP format](#a-bespoke-zip-format)
+  - [Alternate URL schemes considered](#alternate-url-schemes-considered)
+    - [arcp](#arcp)
+    - [pack](#pack)
+- [Stakeholder feedback](#stakeholder-feedback)
+- [Acknowledgements](#acknowledgements)
+
+<!-- /TOC -->
+
+## Proposal
+
+We propose to use the [Web Bundle
+format](https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html),
+without signatures, to give users a format they can create from any website
+they're visiting, share it using existing peer-to-peer sharing apps,
+and then have a peer load it in their browser. The IETF has
+[approved](https://ietf.org/blog/ietf106-highlights/) but not yet created a
+WPACK working group to continue developing this format as of February 2020.
+
+Here's a [demo video](https://youtu.be/xAujz66la3Y) of this use-case for Web
+Bundles (a.k.a. Bundled HTTP Exchanges):
+
+<a href="https://www.youtube.com/watch?v=xAujz66la3Y">
+<img src="https://img.youtube.com/vi/xAujz66la3Y/0.jpg">
+</a>
+
+This explainer doesn't currently propose a standard way to identify all the
+resources from a loaded web page that need to be included in a bundle, or to
+identify a pre-packaged bundle that contains the current page. Instead it
+focuses on how to safely load and navigate within a bundle that's already
+created.
+
+### Relevant structure of a bundle
+
+A bundle is loaded from a "*bundle URL*", and it contains a set of HTTP *items*,
+each with a "*claimed URL*". The items are serialized in a particular order
+within the bundle, and this will affect the performance of loading the bundle
+when the bundle is loaded from a non-random-access medium (like the network),
+but it doesn't affect the semantics of the bundle.
+
+The bundle URL might be a `file://` URL containing private information like the
+user's name, which they might not want exposed to a website they saved for
+offline use.
+
+### Process of loading a bundle
+
+When the browser navigates to an `application/webbundle;v=___` resource, or
+opens a file that the filesystem identifies as that MIME type, it first parses
+the resource enough to pull out the bundle's [primary
+URL](https://wicg.github.io/webpackage/draft-yasskin-wpack-bundled-exchanges.html#name-load-a-bundles-metadata)
+and a set of flags to control loading (not yet defined in the format). One of
```

Comment:
So here do we suppose we'll have some flag inside the format to say trusted vs untrusted?  Should we explicitly mention that the resulting 'trusted' flag set inside the browser may be different from the claimed one, e.g. if unsigned bundle says it wants to be trusted but it's served from different origins?  Or, should the flag only be able to say that it should be treated as 'untrusted' (i.e. downgrading only) even if an unsigned bundle is served from the same origin?

Reply:
Treating it like a suborigin for same-origin untrusted bundle makes sense to me. I'm not so sure about the redirect for 'trusted' but cross-origin bundle, but no strong opinion on that one for now. (Quick cross-checking appreciated for potential subresource loading cases @horo-t @irori @hayatoito)

Reply:
My thinking about 'trusted'-but-cross-origin comes from using the same flag for signed-but-expired packages. We've talked about a need for non-publishers to be able to pick between having a signed-but-expired package redirect vs act-as-unsigned, and the 'trusted' flag seems appropriate for that, and cross-origin is a similar "can't verify the trust" situation.

Reply:
Re: potential subresource loading cases

For subresource loading cases like [this demo](https://docs.google.com/document/d/18fFrURT6xD1GnqJwCx2K3Z81Te9Iv-r2ZsCkOvDH_04/edit#), I don't think we need this 'trusted' flag in the bundle. And also we don’t need primaryUrl/fallbackUrl for subresource web bundle. So we may need to make primaryUrl/fallbackUrl optional.

Reply:
Thanks. It'd be nice if the format can just accommodate both while loading spec part can differ on how to interpret these flags.  While... for navigation case we're probably good to start with this text.